src/allpairs.py

```diff
@@ -0,0 +1,108 @@
+from src.PRM import PRM
+from pathlib import Path
+from src.util import prepare_volume, run_container
+import pandas as pd
+import networkx as nx
+
+__all__ = ['AllPairs']
+
+class AllPairs(PRM):
+    required_inputs = ['nodetypes', 'network']
+
+    @staticmethod
+    def generate_inputs(data, filename_map):
+        """
+        Access fields from the dataset and write the required input files
+        @param data: dataset
+        @param filename_map: a dict mapping file types in the required_inputs to the filename for that type
+        @return:
+        """
+        for input_type in AllPairs.required_inputs:
+            if input_type not in filename_map:
+                raise ValueError("{input_type} filename is missing")
+
+        #Get sources and targets for node input file
+        #Borrowed code from pathlinker.py
+        sources_targets = data.request_node_columns(["sources", "targets"])
+        if sources_targets is None:
+            return False
+        both_series = sources_targets.sources & sources_targets.targets
+        for index,row in sources_targets[both_series].iterrows():
+            warn_msg = row.NODEID+" has been labeled as both a source and a target."
+            warnings.warn(warn_msg)
+
+        #Create nodetype file
+        input_df = sources_targets[["NODEID"]].copy()
+        input_df.columns = ["#Node"]
+        input_df.loc[sources_targets["sources"] == True,"Node type"]="source"
+        input_df.loc[sources_targets["targets"] == True,"Node type"]="target"
+
+        input_df.to_csv(filename_map["nodetypes"],sep="\t",index=False,columns=["#Node","Node type"])
+
+        #This is pretty memory intensive. We might want to keep the interactome centralized.
+        data.get_interactome().to_csv(filename_map["network"],sep="\t",index=False,columns=["Interactor1","Interactor2","Weight"],header=["#Interactor1","Interactor2","Weight"])
+
+
+    @staticmethod
+    def run(nodetypes=None, network=None, output_file=None, singularity=False):
+        """
+        Run AllPairs with Docker
+        @param nodetypes:  input node types with sources and targets (required)
+        @param network:  input network file (required)
+        @param output_file: path to the output pathway file (required)
+        @param singularity: currently inactive, implement later?
```

Comment:
Does it not work with Singularity or has it not been tested with Singularity? I could test it with Singularity if that's the concern. Because you are running the container with `run_container`, it _should_ work with Docker and Singularity.

Reply:
I am unsure by this comment added in the code, I believe it works based on how it looks